### PR TITLE
NayaOne Polkadot Ecosystem Sandbox Application

### PR DESCRIPTION
**Pull request by mvresh**
#### Project Abstract

> Please replace these instructions with a brief description of your project summarising key points (1-2 paragraphs).
>
> If your application is a follow-up to a previous grant, please mention which one in the first line of the abstract and include a link to previous pull requests if applicable.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] A BTC, Ethereum (USDC/DAI) or Polkadot/Kusama (aUSD/USDT) address for the payment of the milestones is provided in the application.
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [x] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@mvresh:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the application. Potentially the research template might be a better fit for your project: https://github.com/w3f/Grants-Program/blob/master/applications/application-template-research.md Have you don’t similar projects for other ecosystem and do you have examples of an Ecosystem Sandbox? If I understand it correctly, you want to make the research open source, but the Dot Hub and the Data Marketplace will be closed-sourced, correct? And are you aware of the Polkadot open-source stack here: https://wiki.polkadot.network/docs/build-open-source ? This might be useful for your project.

**Comment by Noc2** [comment]
Btw. your GitHub organization https://github.com/naya-one links to the following twitter account: https://twitter.com/nayaone Not sure if this is correct.

**Comment by mvresh** [comment]
> Thanks for the application. Potentially the research template might be a better fit for your project: https://github.com/w3f/Grants-Program/blob/master/applications/application-template-research.md Have you don’t similar projects for other ecosystem and do you have examples of an Ecosystem Sandbox? If I understand it correctly, you want to make the research open source, but the Dot Hub and the Data Marketplace will be closed-sourced, correct? And are you aware of the Polkadot open-source stack here: https://wiki.polkadot.network/docs/build-open-source ? This might be useful for your project.

Yes, research application template was used for the application that's submitted in this PR. We have built Digital Sandboxes for the regulators and financial institutions(links provided in the "Additional info" section of the application) although those applications are purpose built for each tenant. With blockchain protocols - this is our first project. We aim to complete the application development of the Polkadot Ecosystem Sandbox (which will be open source as well and more details will be provided as part of the development application) after the research project is complete.

Thanks for sharing the Polkadot stack link. Its definitely helpful. Also, as mentioned in the application, we came across projects/solutions/tools across various use cases that Polkadot and others are already integrated.

**Comment by mvresh** [comment]
> Btw. your GitHub organization https://github.com/naya-one links to the following twitter account: https://twitter.com/nayaone Not sure if this is correct.

Thanks for pointing it out. Following link is the correct one - https://twitter.com/naya_one. Updated in the profile as well.

**Comment by randombishop** [comment]
@mvresh 

Thank you for submitting this very interesting grant application.

If I understand correctly: financial systems and APIs are typically very hard to access and test against, and NayaOne provides an easy way to get development sandboxes ready to go with mock data and testing APIs that replicate what you would get if you plug to the real systems.
From my own experience working in fintech, this can really save a lot of time when trying to develop against classic financial APIs. Very cool!!!

But it's still unclear to me how this helps in blockchain context? Because of the permissionless and open source nature of the crypto space, most blockchains already come with ready to go APIs, different levels of test networks, data endpoints such as https://subquery.network/, and diverse codebases of SDKs and tools, so I am not sure I understand the need to put these into a sandbox? Can you provide one or two examples of concrete things that would end up implemented after the research phase to help me understand better?

Thanks again

**Comment by mvresh** [comment]
> @mvresh
> 
> Thank you for submitting this very interesting grant application.
> 
> If I understand correctly: financial systems and APIs are typically very hard to access and test against, and NayaOne provides an easy way to get development sandboxes ready to go with mock data and testing APIs that replicate what you would get if you plug to the real systems. From my own experience working in fintech, this can really save a lot of time when trying to develop against classic financial APIs. Very cool!!!
> 
> But it's still unclear to me how this helps in blockchain context? Because of the permissionless and open source nature of the crypto space, most blockchains already come with ready to go APIs, different levels of test networks, data endpoints such as https://subquery.network/, and diverse codebases of SDKs and tools, so I am not sure I understand the need to put these into a sandbox? Can you provide one or two examples of concrete things that would end up implemented after the research phase to help me understand better?
> 
> Thanks again

Hi @randombishop . Agree to your point that open source and permissionless nature of the protocols and the tooling like the Subquery makes the testing more direct, open and simpler. As mentioned during our meeting and in the research proposal application, Digital/Innovation Sandbox wrapper to the ecosystem provides a more use case focused experimentation along with integration testing with existing systems and environments within the organisation. Sandbox platforms we've built so far for financial industry participants also have open source technologies and the main objective there is to provide a dedicated environments for large organisations so that their technology teams have one stop shop (with developer environments that can be provisioned) based on the use case(s) and to mock their existing internal workflows together. Larger benefit is with regulators/FIs organising techsprints and challenges for industry collaboration and awareness. 

You can also find the above info in the fig 2 of the application where we created three wrappers around the Ecosystem Sandbox which bring the value to both the Foundation and the ecosystem as well as the other industry participants. 

Another good example of how it benefits the developer ecosystem can be seen with the existing Substrate Cloud Sandbox(https://substrate.io/developers/playground/). It is relevant in making the developer experience better with the templates. Again, use cases for Substrate Playground vary depending on the end user. Taking it to a step further, we'd like to provision a platform where business use case specific sandboxes can be created and re-used by developers both in the ecosystem as well as in other organisations. 

Regarding two concrete examples, Dot Hub members and the Data from the research will further be used to showcase tokenisation and identity use cases(among others) once the Ecosystem Sandbox is complete. It could either be with existing developers in the ecosystem or through a different parachain. 

Hope this clarifies.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the detailed reply, and sorry that I missed the additional Infos at the end. I mark the application as ready for review so that more people take a look at it. But I personally won’t approve it yet. My main concern is that this initial research is mostly beneficial for you, and it’s relatively expensive at this stage. Usually, we ask teams to provide already concrete details about their development work and do this initial research for free.  I understand that your case might be slightly different, but I’m hesitant to support this initial research with 24k. Note others might have a different opinion here.

**Comment by randombishop** [comment]
@mvresh @Noc2 
I concur with @Noc2, I understand the value of your proposal for our ecosystem but the nature of this first deliverable is more educational in nature (tutorials, templates, methodology, reference data and tools) rather than a codebase like the ones we usually fund with clearly defined features and scope.
With that, I would not accept this application in its current version, but would recommend (and/or):
- Submit a Polkadot treasury proposal to fund this first work as educational/prototyping effort.
- Reduce the cost and focus of this first grant on specific features with clearly defined codebase and scope. 
- Make some initial progress then re-apply showing how the deliverable will look like with a "sample"
Thank you!

File: applications/nayaone_polkadot_ecosystem_sandbox_research.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
+# Name of your Research Project1
```
Comment (keeganquigley) [inline applications/nayaone_polkadot_ecosystem_sandbox_research.md:1]: Can you replace the template text with your project name here?

**Review by keeganquigley — CHANGES_REQUESTED** [review]
Hi @mvresh thanks again for the nice call and the application. It is very well written, however I agree with the others that:
- The language is vague and more technical details could be added. For example, "due diligence as-a-service" is mentioned multiple times, but what exactly do these services consist of?
- Can you provide a more concrete vision of what the Dot Hub and sandbox implementations would look like?
- Can you give more detailed descriptions of any languages that would be used, specific tools, etc?
- Regarding your [key points](https://github.com/w3f/Grants-Program/pull/1406/files#diff-472313133804768f60c026cd15739bc27e721e4c23265724822c76ebd643228cR70) of exploration, descriptions such as "use cases" and "tech related info" don't really help me to understand what data will be produced and how it will be useful for anyone other than your clients. Perhaps adding more detail as to how our ecosystem will benefit, and lowering the overall price would help to appease the committee.

Thanks!

**Comment by mvresh** [comment]
@Noc2 @randombishop @keeganquigley 

Thank you all for the comments. Appreciate the feedback as it helps improve the proposal.

Agree that the technical specifics could be made better. We couldn't cover them under this application as the scope and the applicability of the system to be developed are quite broad. Our plan was to include them in the development phase as shown in fig 2 of the application.

Regarding the points about research and educational purposes - major part of this research project involves working with NayaOne tech partners to conduct due diligence on the solutions that enable the adoption of the Polkadot Ecosystem. Within NayaOne existing solution marketplace, we have business data intelligence providers and other market data providers that we work with to provide "due diligence as a service". Typically, before onboarding technology suppliers for our regulatory and institutional clients, it's a step zero to conduct due diligence on the solutions we onboard for them. Hence the same due diligence process check for the Ecosystem Sandbox as well. This service requires our team to access our partner APIs, which we estimated the cost to be between 10k-15k, depending on the number of solutions that are key to the Polkadot Ecosystem.

In case this service seems to be a closed source - Usually, the data collected during due diligence process is publicly available info but our partner solutions provide them in a more accessible and cleaner way. Sample data collected during the process could be company directors' data (based on jurisdiction), past work history, publicly disclosed funding details, team members' professional information, among other details.

Can make necessary changes to the current research proposal to scope the due diligence work into the development phase later (as the benefits are more clearer in this case). Also, the feature is more visible in the Dot Hub component of the ecosystem sandbox.

**Comment by mvresh** [comment]
Hi @Noc2 @randombishop @keeganquigley  thank you all for your inputs on the application. After our discussion last week, we'd like to focus first on the development application and build these components in later stages. Our team is currently defining the application milestones clearly for the dev work. I'll create a new application(development) once the details are finalised. As for the current research application, I'll be closing it. Thanks.

**Pull request closed without merging**